sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/BadRecordException.scala

```diff
@@ -20,6 +20,16 @@ package org.apache.spark.sql.catalyst.util
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.unsafe.types.UTF8String
 
+/**
+ * Exception thrown when the underlying parser returns a partial result of parsing.
+ * @param partialResult the partial result of parsing a bad record.
+ * @param cause the actual exception about why the parser cannot return full result.
+ */
+case class PartialResultException(
```

Comment:
Ur, is this intentional? It looks like `javax.naming.PartialResultException` to me. Not only the same name, but also the semantics.

@cloud-fan . Is this okay? Or, shall we use more distinguishable name like `SparkPartialResultException` instead?

Reply:
We need to carry the `InternalRow` here. I'm fine with the current name, as we don't have Spark prefix in `BadRecordException`

Reply:
Wait .. but let's just rename it if possible .. the cost of renaming is 0 but there are some benefits by that ..

Reply:
I mean, we don't have to standardise the name but let's use another name that doesn't conflict with Java's libraries.